### PR TITLE
40K 10 more

### PR DESCRIPTION
**Pull request by Simisays**
Don't merge this PR before the previous one is merged since this needs the spawn token

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/shard_of_the_nightbringer.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Name:Shard of the Nightbringer
+ManaCost:5 B B B
+Types:Creature C'tan
+PT:8/8
+K:Flying
+T:Mode$ ChangesZone | ValidCard$ Card.Self+wasCast | Destination$ Battlefield | Execute$ TrigLoseLife | TriggerDescription$ Drain Life — When CARDNAME enters the battlefield, if you cast it, target opponent loses half their life, rounded up. You gain life equal to the life lost this way.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shard_of_the_nightbringer.txt:6]: wasCastByYou

File: forge-gui/res/cardsfolder/upcoming/sister_of_silence.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Name:Sister of Silence
+ManaCost:4 U
+Types:Creature Human Knight
+PT:3/3
+K:Flash
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ Psychic Abomination — When CARDNAME enters the battlefield, counter target instant spell, sorcery spell, activated ability, or triggered ability.
+SVar:TrigCounter:DB$ Counter | TargetType$ Spell,Activated,Triggered | TgtPrompt$ Select target spell or ability | ValidTgts$ Card
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/sister_of_silence.txt:7]: For now ValidTgts needs Emblem too

File: forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Name:The Horus Heresy
+ManaCost:3 U B R
+Types:Enchantment Saga
+K:Saga:3:GainControl,DBDraw,DBRepeat
+SVar:GainControl:DB$ GainControl | ValidTgts$ Creature.nonLegendary+OppCtrl | LoseControl$ LeavesPlay,LoseControl | TgtPrompt$ Select target nonlegendary creature an opponent controls to gain control of. | TargetMin$ OneEach | TargetMax$ OneEach | TargetsWithDifferentControllers$ True | SpellDescription$ For each opponent, gain control of up to one target nonlegendary creature that player controls for as long as CARDNAME remains on the battlefield.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt:5]: Is that TargetMin correct?
Also LoseControl shouldn't be both

File: forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+SVar:OneEach:PlayerCountOpponents$Amount
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ X | SpellDescription$ Draw a card for each creature you control but don't own.
+SVar:X:Count$Valid Creature.YouDontOwn+YouCtrl
+SVar:DBRepeat:DB$ RepeatEach | RepeatPlayers$ Player | StartingWithActivator$ True | RepeatSubAbility$ DBChooseCreature | SubAbility$ DBDestroy | SpellDescription$ Starting with you, each player chooses a creature. Destroy each creature chosen this way.
+SVar:DBChooseCreature:DB$ ChooseCard | Defined$ Remembered | Choices$ Creature.RememberedPlayerCtrl | ChoiceTitle$ Choose a creature | Mandatory$ True | SubAbility$ DBChooseLand | RememberChosen$ True
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt:10]: Wrong Choices

Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt:10]: Invalid SubAbility

File: forge-gui/res/cardsfolder/upcoming/the_ruinous_powers.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+Name:The Ruinous Powers
+ManaCost:2 B R
+Types:Enchantment
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigChoose | TriggerDescription$ At the beginning of your upkeep, choose an opponent at random. Exile the top card of that player's library. Until end of turn, you may play that card and you may spend mana as though it were mana of any color to cast it. When you cast a spell this way, its owner loses life equal to its mana value.
+SVar:TrigChoose:DB$ ChoosePlayer | Defined$ You | Choices$ Opponent | Random$ True | SubAbility$ DBExile
+SVar:DBExile:DB$ Dig | Defined$ ChosenPlayer | DigNum$ 1 | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect 
+SVar:DBEffect:DB$ Effect | StaticAbilities$ STPlay | Triggers$ TriggerCastDoM | RememberObjects$ Remembered
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_ruinous_powers.txt:7]: You can just do `ExileOnMoved$ Exile` instead of using Triggers$

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_ruinous_powers.txt:7]: but the second real trigger should be made part of the effect

File: forge-gui/res/cardsfolder/upcoming/the_ruinous_powers.txt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+SVar:STPlay:Mode$ Continuous | MayPlay$ True | MayPlayIgnoreColor$ True | EffectZone$ Command | Affected$ Card.IsRemembered+nonLand | AffectedZone$ Exile | Description$ Until end of turn, you may cast that card and you may spend mana as though it were mana of any color to cast it.
+SVar:TriggerCastDoM:Mode$ SpellCast | ValidCard$ Card.IsRemembered | TriggerZones$ Command | Execute$ TrigRemoveSelf | Static$ True
+SVar:TrigRemoveSelf:DB$ ChangeZone | Defined$ Self | Origin$ Command | Destination$ Exile
+T:Mode$ SpellCast | ValidCard$ Card.IsRemembered | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigLoseLife | Secondary$ True | TriggerDescription$ At the beginning of your upkeep, choose an opponent at random. Exile the top card of that player's library. Until end of turn, you may play that card and you may spend mana as though it were mana of any color to cast it. When you cast a spell this way, its owner loses life equal to its mana value.
+SVar:TrigLoseLife:DB$ LoseLife | Defined$ TriggeredCardOwner | LifeAmount$ X | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:TriggeredCard$CardManaCost
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_ruinous_powers.txt:14]: Use `TriggeredStackInstance$CardManaCostLKI`

File: forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt (hunk 3 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt:5]: I've found that with Blatant Thievery. I haven't tested multiplayer yet but it works 1v1

File: forge-gui/res/cardsfolder/upcoming/thunderhawk_gunship.txt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+Name:Thunderhawk Gunship
+ManaCost:6
+Types:Artifact Vehicle
+PT:6/6
+K:Flying
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigToken | TriggerDescription$ When CARDNAME enters the battlefield, create two 2/2 white Astartes Warrior creature tokens with vigilance.
+SVar:TrigToken:DB$ Token | TokenScript$ W_2_2_astartes_warrior_vigilance | TokenAmount$ 2
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigPump | TriggerDescription$ Whenever CARDNAME attacks, attacking creatures you control gain flying until end of turn.
+SVar:TrigPump:DB$ PumpAll | ValidCards$ Creature.attacking | KW$ Flying
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/thunderhawk_gunship.txt:9]: Missing YouCtrl

File: forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt (hunk 3 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/the_horus_heresy.txt]: But that card doesn't say "up to"

File: forge-gui/res/cardsfolder/upcoming/the_ruinous_powers.txt [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+SVar:DBExile:DB$ Dig | Defined$ ChosenPlayer | DigNum$ 1 | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect 
+SVar:DBEffect:DB$ Effect | StaticAbilities$ STPlay | Triggers$ TriggerCastDoM | ExileOnMoved$ Exile | RememberObjects$ Remembered
+SVar:STPlay:Mode$ Continuous | MayPlay$ True | MayPlayIgnoreColor$ True | EffectZone$ Command | Affected$ Card.IsRemembered+nonLand | AffectedZone$ Exile | Description$ Until end of turn, you may cast that card and you may spend mana as though it were mana of any color to cast it.
+SVar:TriggerCastDoM:Mode$ SpellCast | ValidCard$ Card.IsRemembered | ValidActivatingPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigLoseLife | Secondary$ True | TriggerDescription$ At the beginning of your upkeep, choose an opponent at random. Exile the top card of that player's library. Until end of turn, you may play that card and you may spend mana as though it were mana of any color to cast it. When you cast a spell this way, its owner loses life equal to its mana value.
+SVar:TrigLoseLife:DB$ LoseLife | Defined$ TriggeredCardOwner | LifeAmount$ X | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_ruinous_powers.txt:10]: No need for Cleanup, Effect will disappear anyway

**Comment by Simisays** [comment]
Is this PR good to go or is something  still lacking ?

**Comment by tool4ever** [comment]
Hmn, I was waiting for Northmoc since he usually finds something I've missed but looks like he might be too busy

**Comment by Simisays** [comment]
Yeah that's what I've thought. I did change Sister of Silence to make it only target instants and sorceries. FYI

File: forge-gui/res/cardsfolder/upcoming/shard_of_the_nightbringer.txt [hunk 9]
```diff
@@ -0,0 +1,11 @@
+Name:Shard of the Nightbringer
+ManaCost:5 B B B
+Types:Creature C'tan
+PT:8/8
+K:Flying
+T:Mode$ ChangesZone | ValidCard$ Card.Self+wasCastByYou | Destination$ Battlefield | Execute$ TrigLoseLife | TriggerDescription$ Drain Life — When CARDNAME enters the battlefield, if you cast it, target opponent loses half their life, rounded up. You gain life equal to the life lost this way.
+SVar:TrigLoseLife:DB$ LoseLife | ValidTgts$ Opponent | LifeAmount$ X | SubAbility$ DBGainLife
+SVar:DBGainLife:DB$ GainLife | Defined$ You | LifeAmount$ AFLifeLost
+SVar:X:Count$TargetedLifeTotal/HalfUp
+DeckHas:Ability$LifeGain
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/shard_of_the_nightbringer.txt:10]: missing `SVar:AFLifeLost:Number$0` ?

File: forge-gui/res/cardsfolder/upcoming/shard_of_the_void_dragon.txt [hunk 10]
```diff
@@ -0,0 +1,12 @@
+Name:Shard of the Void Dragon
+ManaCost:4 B B B
+Types:Creature C'tan
+PT:7/7
+K:Flying
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ DBSacrifice | TriggerZones$ Battlefield | TriggerDescription$ Sphere of the Void Dragon — Whenever Shard of the Void Dragon attacks, each opponent sacrifices a nonland permanent.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/shard_of_the_void_dragon.txt:6]: Whenever CARDNAME

File: forge-gui/res/cardsfolder/upcoming/shard_of_the_void_dragon.txt [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+SVar:DBSacrifice:DB$ Sacrifice | Amount$ 1 | SacValid$ Permanent.nonLand | Defined$ Opponent 
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard,Exile | ValidCard$ Artifact | Execute$ TrigCounter | TriggerDescription$ Matter Absorption — Whenever an artifact is put into a graveyard from the battlefield or is put into exile from the battlefield, put two +1/+1 counters on CARDNAME.
+SVar:TrigCounter:DB$ PutCounter | Defined$ Self | CounterType$ P1P1 | CounterNum$ 2
+DeckHas:Ability$Sacrifice|Counters
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/shard_of_the_void_dragon.txt:10]: add `DeckHints:Type$Artifact|Clue|Treasure` or something like that? Little artifacts that sac themselves would really fuel this one

File: forge-gui/res/cardsfolder/upcoming/sister_of_silence.txt [hunk 12]
```diff
@@ -0,0 +1,8 @@
+Name:Sister of Silence
+ManaCost:4 U
+Types:Creature Human Knight
+PT:3/3
+K:Flash
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ Psychic Abomination — When CARDNAME enters the battlefield, counter target instant spell, sorcery spell, activated ability, or triggered ability.
+SVar:TrigCounter:DB$ Counter | TargetType$ Spell,Activated,Triggered | TgtPrompt$ Select target instant spell, sorcery spell, activated ability, or triggered ability | ValidTgts$ Instant,Sorcery,Emblem
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/sister_of_silence.txt:7]: This ValidTgts seems wrong to me... what about activated abilities of permanents?

File: forge-gui/res/cardsfolder/upcoming/sloppity_bilepiper.txt [hunk 13]
```diff
@@ -0,0 +1,10 @@
+Name:Sloppity Bilepiper
+ManaCost:3 R
+Types:Creature Demon
+PT:3/3
+A:AB$ Effect | PrecostDesc$ Jolly Gutpipes — | Cost$ 2 T Sac<1/Creature> | StaticAbilities$ GrantCascade | Triggers$ ExileEffect | SpellDescription$ The next creature spell you cast this turn has cascade. (When you cast your next creature spell, exile cards from the top of your library until you exile a nonland card that costs less. You may cast it without paying its mana cost. Put the exiled cards on the bottom of your library in a random order.)
+SVar:GrantCascade:Mode$ Continuous | EffectZone$ Command | Affected$ Card.Creature+YouCtrl | AffectedZone$ Stack | Execute$ ExileEff | AddKeyword$ Cascade | Description$ The next noncreature spell you cast this turn has cascade.
+SVar:ExileEffect:Mode$ SpellCast | EffectZone$ Command | Valid$ Card.Creature+YouCtrl  | AffectedZone$ Stack | Execute$ RemoveEffect | Static$ True
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/sloppity_bilepiper.txt:7]: spacing

File: forge-gui/res/cardsfolder/upcoming/sloppity_bilepiper.txt [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+SVar:RemoveEffect:DB$ ChangeZone | Origin$ Command | Destination$ Exile | Defined$ Self
```
[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/sloppity_bilepiper.txt]: I don't think so, there's nothing to remember

**Review by tool4ever — APPROVED** [review]

**Pull request merged**